### PR TITLE
sub_ids and different contexts

### PR DESCRIPTION
**Pull request by adeinega**
Suggested changes for https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/178

### CODE REVIEW

**Comment by fimbault** [comment]
Not sure "subject_type_identifiers" is best (when I read that, what my brain remembers is identifiers, and that's the opposite of what #177 is targeting). 

In 2.2 this is already encapsulated within a "subject" object. In section 9, that's not the case (so we need to explicit that it relates to sub / subject) 
-> "sub_ids_types" or "sub_types"?

**Comment by adeinega** [comment]
@fimbault, I tried to be consistent with https://www.ietf.org/archive/id/draft-ietf-secevent-subject-identifiers-06.txt and called it _subject_type_identifiers_, not "sub_typ_ids" or somehow else... even though, as you know, there is a clear tendency to use short names, e.g. "sub", "kid" and so on and so forth.

IMO, OpenID Connect (in general) and OpenID Connect Discovery (in particular), very consistent with naming and uses "subject_types_supported" and we may also consider using it. What I mean is
1. subject_types_requested for section 2
2. subject_types_supported for section 9

**Comment by fimbault** [comment]
Yes I like that

**Comment by adeinega** [comment]
I've replaced _subject_type_identifiers_ by _subject_types_supported_ which also means is that if you wish to go this route, you would also need to replace at least

1. _key_proofs_ by _key_proofs_supported_
2.  and _assertions_ by _assertion_supporrted_

in section Discovery for the sake of consistency.

These changes, as well as ones in https://github.com/ietf-wg-gnap/gnap-core-protocol/pull/181, aren't ideal and, I believe there will be lots of discussions about subject type identifiers. This is just one little step towards clarity.

**Comment by fimbault** [comment]
After reviewing with the editors, we suggest that we use "subject_types" for section 2 (to be close to secevents) and "subject_types_supported" for section 9.

If additional changes for names are suggested for consistency (keys, assertions), a separate PR is probably best for tracking changes

**Comment by adeinega** [comment]
I've added these changes and created a separate https://github.com/ietf-wg-gnap/gnap-core-protocol/pull/194 Discovery.

File: draft-ietf-gnap-core-protocol.md [hunk 1]
```diff
@@ -4094,9 +4094,9 @@ key_proofs (array strings)
           values of the `proof` field of the 
           [key section](#key-format) of the request.
 
-sub_ids (array of strings)
+subject_types_supported (array of strings)
 : OPTIONAL. A list of the AS's supported
-          identifiers. The values of this list correspond to possible values
+          identifier types. The values of this list correspond to possible values
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:4099]: To be particularly pedantic:

```suggestion
          subject identifier types. The values of this list correspond to possible values
```

**Comment by jricher** [comment]
I just realized that this will also need an updated entry in the Changelog since it's a substantive change.

**Comment by adeinega** [comment]
https://github.com/ietf-wg-gnap/gnap-core-protocol/pull/184/commits/a6503423d5c600d3ca612d6760e3551f28c37a6c also fixes a couple of other examples I haven't noticed initially.

**Comment by jricher** [comment]
@adeinega This PR is fine, but in the future, please do not add additional edits after the editors have marked it "Pending Merge". If it's not ready we can back it out of that state, we just want to make sure any WG members have a chance to see the changes as they go in. Thanks!

**Comment by fimbault** [comment]
since SECEVENT has now called the types "format", just wondering if that should be aligned

**Comment by jricher** [comment]
@fimbault I saw that also, and I think we should do a sweep to align them all in one go. My read of the response to this PR is that there's support for changing the terms in this direction (adding _supported and the like). So we might want to hold off on this PR specifically and do all the changes at once. This would affect things I have in #211 and others, so the editors basically just need to rack and stack these to get things fixed throughout the document.

**Comment by jricher** [comment]
(alternatively we can fold in this PR just to change things again with another PR that aligns the terms, which is also reasonable)

**Comment by adeinega** [comment]
Yes... it's quite interesting that the most recent https://datatracker.ietf.org/doc/html/draft-ietf-secevent-subject-identifiers-07 started to use `format` instead of `subject_type`. I believe GNAP should be aligned with _Subject Identifiers for Security Event Tokens_ but it also means it needs to use something a bit more descriptive than something like Discovery parameters' `format_types_supported` or more lengthy `subject_identifier_formats_supported`.

Who knows what next versions of draft-ietf-secevent-subject-identifiers would use in the future? `fmt`?

**Comment by agropper** [comment]
I'm trying to understand how we're using sub_id. See https://mailarchive.ietf.org/arch/msg/txauth/0o3lPPADL5AY84XmqEsWsxU4j2E/

**Comment by fimbault** [comment]
@adeinega good remarks. I don't know why they settled for format, but whatever term they choose, I guess we could ask them to stabilize that once and for all, and avoid forcing us do an additional mapping. 
For discovery, I would suggest `subject_formats_supported`?

**Comment by jricher** [comment]
Accepting as is, will update all SECEVENT references together in a future PR.

**Pull request merged**